release-blocking-jobs.md

```diff
@@ -0,0 +1,50 @@
+# Release Blocking Jobs
+
+Here is a list of jobs that should block release if not healthy.
+
+If you want to add a release-branch job, please add them to sig-release-x.x-all dashboard first,
+and show the test is not flaky, then the release team can decide if the test should be added
```

Comment:
I know @spiffxp was working on test flakiness criteria before letting jobs block a presubmit or release. May be worth linking to that here. If thats not ready yet, may be indicate a minimum threshold of stability before we allow a job to be release blocking

Reply:
I am fine with us merging as is and refining the criteria later